extendr-api/src/robj/tests.rs

```diff
@@ -153,10 +153,20 @@ fn test_try_from_robj() {
 
         assert_eq!(<Vec::<i32>>::try_from(Robj::from(1)), Ok(vec![1]));
         assert_eq!(<Vec::<f64>>::try_from(Robj::from(1.)), Ok(vec![1.]));
+        assert_eq!(<Vec::<Bool>>::try_from(Robj::from(TRUE)), Ok(vec![TRUE]));
+        assert_eq!(<Vec::<u8>>::try_from(Robj::from(0_u8)), Ok(vec![0_u8]));
+
+        assert_eq!(<&[i32]>::try_from(Robj::from(1)), Ok(&[1][..]));
```

Comment:
What happens when there is a type mismatch, e.g.:
```rust
<&[i32]>::try_from(Robj::from(1.5))
```
Should there be assertion statements for these kinds of cases?

Reply:
Tests added,